hubblestack/files/hubblestack_nova/misc.py

```diff
@@ -509,6 +509,256 @@ def check_core_dumps(reason=''):
+    	client_alive_count_max = _execute_shell_command("grep \"^ClientAliveCountMax\" /etc/ssh/sshd_config | awk '{print $NF}'").strip()
+    	if client_alive_count_max != '' and int(client_alive_count_max) < 3:
+       	    return True
+    	else:
+            return "ClientAliveCountMax value should be less than 3"
+    else:
+    	return "ClientAliveInterval value should be less than 300"
+
+
+def check_unowned_files(reason=''):
+    '''
+    Ensure no unowned files or directories exist
+    '''
+
+    unowned_files = _execute_shell_command("df --local -P | awk 'NR!=1 {print $6}' | xargs -I '{}' find '{}' -xdev -nouser 2>/dev/null").strip()
+    unowned_files = unowned_files.split('\n') if unowned_files != "" else []
+    # The command above only searches local filesystems, there may still be compromised items on network mounted partitions.
+    # Following command will check each partition for unowned files
+    unowned_partition_files = _execute_shell_command("mount | awk '{print $3}' | xargs -I '{}' find '{}' -xdev -nouser 2>/dev/null").strip()
+    unowned_partition_files = unowned_partition_files.split('\n') if unowned_partition_files != "" else []
+    unowned_files = unowned_files + unowned_partition_files
+    if unowned_files == []:
+        return True
+
+    return str(list(set(unowned_files)))
+
+
+def check_ungrouped_files(reason=''):
+    '''
+    Ensure no ungrouped files or directories exist
+    '''
+
+    ungrouped_files = _execute_shell_command("df --local -P | awk 'NR!=1 {print $6}' | xargs -I '{}' find '{}' -xdev -nogroup 2>/dev/null").strip()
+    ungrouped_files = ungrouped_files.split('\n') if ungrouped_files != "" else []
+    # The command above only searches local filesystems, there may still be compromised items on network mounted partitions.
+    # Following command will check each partition for unowned files
+    ungrouped_partition_files = _execute_shell_command("mount | awk '{print $3}' | xargs -I '{}' find '{}' -xdev -nogroup 2>/dev/null").strip()
+    ungrouped_partition_files = ungrouped_partition_files.split('\n') if ungrouped_partition_files != "" else []
+    ungrouped_files = ungrouped_files + ungrouped_partition_files
+    if ungrouped_files == []:
+        return True
+
+    return str(list(set(ungrouped_files)))
+
+
+def check_all_users_home_directory(reason=''):
+    '''
+    Ensure all users' home directories exist
+    '''
+
+    users_uids_dirs = _execute_shell_command("cat /etc/passwd | awk -F: '{ print $1 \" \" $3 \" \" $6 }'").strip()
+    users_uids_dirs = users_uids_dirs.split('\n') if users_uids_dirs != "" else []
+    error = []
+    for user_data in users_uids_dirs:
+        user_uid_dir = user_data.strip().split(" ")
+        if len(user_uid_dir) < 3:
+                user_uid_dir = user_uid_dir + ['']*(3-len(user_uid_dir))
+        if user_uid_dir[1].isdigit():
+            if int(user_uid_dir[1]) >= 1000 and not os.path.isdir(user_uid_dir[2]) and user_uid_dir[0] is not "nfsnobody":
```

Comment:
On some other OS like Amazon Linux, we need value as 500 rather than 1000, can you please take this as parameter as max_system_uid and use it here. So that we can use this function for Amazon Linux as well in future.